geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z2IdxStrategy.scala

```diff
@@ -175,16 +176,24 @@ object Z2IdxStrategy extends StrategyProvider {
   val Z2_ITER_PRIORITY = 21
   val FILTERING_ITER_PRIORITY = 25
 
+  override protected def statsBasedCost(sft: SimpleFeatureType,
+                                        filter: QueryFilter,
+                                        transform: Option[SimpleFeatureType],
+                                        stats: GeoMesaStats): Option[Long] = {
+    filter.singlePrimary match {
+      case None => Some(Long.MaxValue)
+      // add one so that we prefer the z3 index even if geometry is the limiting factor, resulting in the same count
+      case Some(f) => stats.getCount(sft, f, exact = false).map(c => if (c == 0L) 0L else c + 1L)
+    }
+  }
+
   /**
-    * TODO update description
-    * Gets the estimated cost of running the query. Currently, cost is hard-coded to sort between
-    * strategies the way we want. Z2 should be more than id lookups (at 1), high-cardinality attributes (at 1)
-    * and less than STidx (at 400) and unknown cardinality attributes (at 999).
-    *
-    * Eventually cost will be computed based on dynamic metadata and the query.
+    * More than id lookups (at 1), high-cardinality attributes (at 1), z3 (at 200).
+    * Less than spatial-only z3 (at 401), unknown cardinality attributes (at 999).
```

Comment:
yeah, it will use z2 for spatial only queries. because it's not binned by week, it will be a lot faster.